File: library/api/src/main/java/org/cru/godtools/api/LanguagesApi.java
package org.cru.godtools.api;

import org.ccci.gto.android.common.jsonapi.model.JsonApiObject;
import org.ccci.gto.android.common.jsonapi.retrofit2.JsonApiParams;
import org.cru.godtools.model.Language;

import androidx.annotation.NonNull;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.QueryMap;

public interface LanguagesApi {
    String PATH_LANGUAGES = "languages";

    @GET(PATH_LANGUAGES)
    Call<JsonApiObject<Language>> list(@QueryMap @NonNull JsonApiParams params);
}

File: library/api/src/main/java/org/cru/godtools/api/LanguagesApi.kt
package org.cru.godtools.api

import org.ccci.gto.android.common.jsonapi.model.JsonApiObject
import org.ccci.gto.android.common.jsonapi.retrofit2.JsonApiParams
import org.cru.godtools.model.Language
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.QueryMap

private const val PATH_LANGUAGES = "languages"

interface LanguagesApi {
    @GET(PATH_LANGUAGES)
    suspend fun list(@QueryMap params: JsonApiParams): Response<JsonApiObject<Language>>
}
